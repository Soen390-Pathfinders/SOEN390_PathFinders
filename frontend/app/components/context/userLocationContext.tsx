//Context to share the user location across the app for differentcomponents can access it
import React, { createContext, useContext, useState } from "react";

// Define the strcutre of the context
interface LocationContextType {
  userLocation: { latitude: number; longitude: number }; // stores the current user location
  updateUserLocation: (latitude: number, longitude: number) => void; // updates the user current location
}

// Create the context for the user location , the default value is undefined
const LocationContext = createContext<LocationContextType | undefined>(
  undefined
);

//defines LocationProvider , this needs to wrap the whole app, all the children of the  app root will have access to the user location
export const LocationProvider: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  const [userLocation, setUserLocation] = useState({
    latitude: -73.57907171854552,
    longitude: 45.49749147752672,
  }); // initial location is Hall building  (45.49749147752672, -73.57907171854552)

  // Function to update the location
  const updateUserLocation = (latitude: number, longitude: number) => {
    setUserLocation({ latitude, longitude });
  };

  return (
    <LocationContext.Provider value={{ userLocation, updateUserLocation }}>
      {children}
    </LocationContext.Provider>
  );
};

// hook to use the current user location context
export const useLocation = () => {
  const context = useContext(LocationContext);
  if (!context) {
    throw new Error("useLocation must be used within a LocationProvider");
  }
  return context;
};
export default useLocation;
