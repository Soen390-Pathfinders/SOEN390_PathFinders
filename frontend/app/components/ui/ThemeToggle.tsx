import { View, TouchableOpacity, Image } from "react-native";
import { getStyles } from "../../styles";
import useTheme from "../../hooks/useTheme";

export default function ThemeToggle() {
  const { theme, toggleTheme } = useTheme(); // Use the custom hook
  const styles = getStyles(theme); // Get styles based on theme

  const handleToggleTheme = () => {
    const newTheme = theme === "light" ? "dark" : "light";
    toggleTheme(newTheme);
  };

  return (
    <View style={styles.container}>
      <View //style={styles.iconRowView}
      >
        <TouchableOpacity onPress={handleToggleTheme}>
          <Image
            style={{ height: 40, width: 40 }}
            source={
              theme === "dark"
                ? require("../../../assets/images/brightness.png")
                : require("../../../assets/images/sleep-mode.png")
            }
            tintColor={theme === "dark" ? "#000" : "#fff"} // Invert the color based on theme
          />
        </TouchableOpacity>
      </View>
    </View>
  );
}
