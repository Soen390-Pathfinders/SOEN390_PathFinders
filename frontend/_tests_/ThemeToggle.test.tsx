import React from "react";
import renderer from "react-test-renderer";
import ThemeToggle from "@/app/components/ui/ThemeToggle";
import { ThemeProvider } from "@/app/components/context/ThemeContext";


jest.mock('@react-native-async-storage/async-storage', () => ({
  setItem: jest.fn(),
  getItem: jest.fn(() => Promise.resolve('light')),
  removeItem: jest.fn(),
  clear: jest.fn(),
}));

test("ThemeToggle renders correctly", () => {
  const tree = renderer.create(
    <ThemeProvider>
      <ThemeToggle />
    </ThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});