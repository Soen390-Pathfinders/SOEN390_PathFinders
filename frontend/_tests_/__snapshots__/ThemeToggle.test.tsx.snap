// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/images/sleep-mode.png",
          }
        }
        style={
          {
            "height": 40,
            "width": 40,
          }
        }
        tintColor="#fff"
      />
    </View>
  </View>
</View>
`;
